### ISSUE TITLE
Undefined error when using apollo-boost/apollo-cache-inmemory.

### ISSUE THREAD

**Issue by joelmandell** [apollographql/apollo-client#4218]
Hey! I am using apollo-cache-inmemory 1.3.11, tried backwards to 1.3.4. But I get same error as screenshot below shows:

![image](https://user-images.githubusercontent.com/109020/49528179-c3a4ec00-f8b3-11e8-98d2-03d4a9a7b586.png)

What am I doing wrong?

**Comment by joelmandell** [comment]
I would like to clarify, that this happens when I import apollo-boost.

Anyway, I solved this by importing apollo-cache-inmemory, and those requirements by myself instead of using apollo-boost. And then I do not get this exception.

**Comment by benjamn** [comment]
Is the `graphql` package installed in your app? That's where `QueryDocumentKeys` is supposed to be coming from (specifically, from the `graphql/language/visitor` module).

**Comment by joelmandell** [comment]
Yes, it is. This is my package.json:
```json
  "dependencies": {
    "apollo-boost": "^0.1.22",
    "apollo-cache-inmemory": "1.4.0-alpha.1",
    "apollo-client": "^2.4.7",
    "apollo-link-http": "^1.5.7",
    "bootstrap": "^3.3.7",
    "graphql": "^14.0.2",
    "graphql-tag": "^2.10.0",
    "react": "^16.0.0",
    "react-apollo": "^2.3.2",
    "react-bootstrap": "^0.31.5",
    "react-dom": "^16.0.0",
    "react-router-bootstrap": "^0.24.4",
    "react-router-dom": "^4.2.2",
    "react-scripts": "1.0.17",
    "rimraf": "^2.6.2"
  }
```

**Comment by joelmandell** [comment]
Is the graphql package supposed to be installed in node_modules\apollo-boost\node_modules\?
As it is now, folder structure is like this:
![image](https://user-images.githubusercontent.com/109020/49657561-4064d100-fa40-11e8-8e1a-a485380fecd7.png)

I removed the node_modules from ClientApp folder, and did "npm install" from scratch. Still same problem.
Is it suppose to contain graphql in itself? E.g. node_modules\apollo-boost\node_modules\graphql?

**Comment by arist0tl3** [comment]
What a goose chase, lol.

@joelmandell If you update your `react-scripts` to ^1.1.2, I think you will see this issue resolved. Relevant comment and thread: https://github.com/graphql/graphql-js/issues/1248#issuecomment-378086769

tl;dr: `react-scripts` prior to 1.1.2 wasn't respecting .mjs files, leading to an undefined export in this case.

**Comment by joelmandell** [comment]
Yea that is probarly it. This was a project based on an .net core template.
